rbac/management/role/definer.py

```diff
@@ -274,3 +290,105 @@ def delete_permission(permission: Permission):
             dual_write_handler.replicate_update_system_role()
         else:
             dual_write_handler.replicate_new_or_updated_role(role)
+
+
+def _create_single_platform_role(access_type, scope, policy_service, public_tenant):
+    """Create a single platform role with the given parameters."""
+    uuid = platform_v2_role_uuid_for(access_type, scope, policy_service)
+
+    role_name = f"{access_type.value.capitalize()} {scope.name.lower()} Platform Role"
+    description = f"Platform default role for {access_type.value} access at {scope.name.lower()} scope"
+
+    platform_role, created = PlatformRoleV2.objects.update_or_create(
+        uuid=uuid,
+        defaults={
+            "name": role_name,
+            "description": description,
+            "tenant": public_tenant,
+        },
+    )
+
+    if created:
+        logger.info("Created platform role: %s", role_name)
+    else:
+        logger.info("Updated platform role: %s", role_name)
+
+    return platform_role
+
+
+def _seed_v2_role_from_v1(v1_role, display_name, description, public_tenant, platform_roles, resource_service):
+    """Create or update V2 role from V1 role during seeding."""
+    try:
+        v2_role, v2_created = SeededRoleV2.objects.update_or_create(
+            uuid=v1_role.uuid,
+            defaults={
+                "name": display_name,
+                "description": description,
+                "tenant": public_tenant,
+                "v1_source": v1_role,
+            },
+        )
+        if v2_created:
+            logger.info("Created V2 system role %s.", display_name)
+        else:
+            logger.info("Updated V2 system role %s.", display_name)
+        v2_role.permissions.clear()
+        v1_permissions = [access.permission for access in v1_role.access.all()]
+        if v1_permissions:
+            v2_role.permissions.set(v1_permissions)
+            logger.info("Added %d permissions to V2 role %s.", len(v1_permissions), display_name)
+        scope = resource_service.scope_for_role(v1_role)
+
+        # Clear parents first since scope may have changed since previous seeding
+        v2_role.parents.clear()
+        platform_role = platform_roles[(DefaultAccessType.USER, scope)]
+        if v1_role.platform_default:
+            platform_role.children.add(v2_role)
+            logger.info("Added %s as child of platform role %s", display_name, platform_role.name)
+        else:
+            platform_role.children.discard(v2_role)
```

Comment:
This seems redundant because we already clear the parents: v2_role.parents.clear()